include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -220,7 +220,8 @@ class geometry_to_segment_or_box
         comparable_return_type cd_min1(0);
         point_iterator_type pit_min;
         seg_or_box_iterator_type it_min1 = seg_or_box_points.begin();
-        seg_or_box_iterator_type it_min2 = ++seg_or_box_points.begin();
+        seg_or_box_iterator_type it_min2 = it_min1;
+        ++it_min2;
```

Comment:
Actually the above may not be true. It's possible that only `std::vector<>::const_reverse_iterator` fails on Solaris this way. Either way, the code should compile now.